File: PHP/Program/SmartDevice.php
<?php
require_once 'Device.php';

class SmartDevice extends Device {
    private $os = '';
    private $storage = 0;
    private $ram = 0;

    public function __construct($id = 0, $name = '', $price = 0, $stock = 0, $photo = '', $brand = '', $warranty = 0, $power = 0, $os = '', $storage = 0, $ram = 0) {
        parent::__construct($id, $name, $price, $stock, $photo, $brand, $warranty, $power);
        $this->os = $os;
        $this->storage = $storage;
        $this->ram = $ram;
    }

    public function setOS ($os) { $this->os = $os; }
    public function getOS () { return $this->os; }

    public function setStorage ($storage) { $this->storage = $storage; }
    public function getStorage () { return $this->storage; }

    public function setRAM ($ram) { $this->ram = $ram; }
    public function getRAM () { return $this->ram; }
}
?>

File: Python/Program/SmartDevice.py
from Device import Device

class SmartDevice (Device):
    __os = ""
    __storage = 0
    __ram = 0

    def __init__(self, id=0, name="", price=0, stock=0, brand="", warranty=0, power=0, os="", storage=0, ram=0):
        super().__init__(id, name, price, stock, brand, warranty, power)
        self.__os = os
        self.__storage = storage
        self.__ram = ram
    
    def setOS(self, os):
        self.__os = os

    def getOS(self):
        return self.__os
    
    def setStorage(self, storage):
        self.__storage = storage

    def getStorage(self):
        return self.__storage
    
    def setRAM(self, ram):
        self.__ram = ram

    def getRAM(self):
        return self.__ram